stackslib/src/chainstate/nakamoto/miner.rs

```diff
@@ -521,6 +532,24 @@ impl NakamotoBlockBuilder {
             .block_limit()
             .expect("Failed to obtain block limit from miner's block connection");
 
+        let mut soft_limit = None;
+        if let Some(percentage) = settings
+            .mempool_settings
+            .tenure_cost_limit_per_block_percentage
+        {
+            if percentage < 100 {
+                let mut limit = block_limit.clone();
+                if limit.divide(100).is_ok() {
+                    limit.multiply(percentage.into()).expect(
+                        "BUG: failed to multiply by {percentage} when previously divided by 100",
+                    );
+                    soft_limit = Some(limit);
+                }
+            }
+        }
+
+        builder.soft_limit = soft_limit;
```

Comment:
Am I reading this wrong, or will the new soft limit always be `BLOCK_LIMIT * percentage`? Or does `tenure_tx.block_limit` return the _remaining_ block limit?

Reply:
Gah. Yes you are right. I have changed this code so many times today, and each time I miss something. :P Will fix.

Reply:
Actually..no I think this does work correctly because I think the block limit reduces each time within the call to create the new tenure_tx

```
        // Nakamoto must load block cost from parent if this block isn't a tenure change.
        // If this is a tenure-extend, then the execution cost is reset.
        let initial_cost = if new_tenure || tenure_extend {
            ExecutionCost::zero()
        } else {
            let parent_cost_total =
                Self::get_total_tenure_cost_at(chainstate_tx.as_tx(), &parent_index_hash)?
                    .ok_or_else(|| {
                        ChainstateError::InvalidStacksBlock(format!(
                    "Failed to load total tenure cost from parent. parent_stacks_block_id = {}",
                    &parent_index_hash
                ))
                    })?;
            parent_cost_total
        };
```

Reply:
Ok awesome! Yep, makes sense, just wanted to double check.